tracing-core/src/field.rs

```diff
@@ -648,6 +648,11 @@ impl Field {
     pub fn name(&self) -> &'static str {
         self.fields.names[self.i]
     }
+
+    /// Returns the index of this field in its [`FieldSet`].
+    pub fn index(&self) -> usize {
```

Comment:
It would be great to see an integration test for this function.

Reply:
Added. But I am not sure if I added it in the right place.

Reply:
I think that the tests should be separated out. They should also be placed in `tracing-core` which is where this functionality is.

It may be enough to construct a `FieldSet` and then get a field out of it and check that the index is the expected one for each field. We don't need to go through a collector or dispatcher (so we don't need to use `tracing::event!` or `tracing::span!` macros.

Reply:
Got it. Updated. Thanks!